### PR TITLE
symex_target_equation: use ranged for

### PR DESCRIPTION
**Pull request by kroening**
improve readability by using a ranged for over the steps

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-symex/symex_target_equation.h [hunk 1]
```diff
@@ -349,21 +349,17 @@ class symex_target_equationt:public symex_targett
...
   {
     std::size_t i=0;
-    for(SSA_stepst::const_iterator
-        it=SSA_steps.begin();
-        it!=SSA_steps.end(); it++)
-      if(it->is_assert())
+    for(const auto &step : SSA_steps)
+      if(step.is_assert())
         i++;
     return i;
```
Comment (tautschnig) [inline src/goto-symex/symex_target_equation.h]: How about `return std::count_if(SSA_steps.begin(), SSA_steps.end(), [](const SSA_stept &step) { return step.is_assert(); });`?

File: src/goto-symex/symex_target_equation.h [hunk 2]
```diff
@@ -349,21 +349,17 @@ class symex_target_equationt:public symex_targett
...
   {
     std::size_t i=0;
-    for(SSA_stepst::const_iterator
-        it=SSA_steps.begin();
-        it!=SSA_steps.end(); it++)
-      if(it->ignore)
+    for(const auto &step : SSA_steps)
+      if(step.ignore)
         i++;
     return i;
```
Comment (tautschnig) [inline src/goto-symex/symex_target_equation.h]: How about `return std::count_if(SSA_steps.begin(), SSA_steps.end(), [](const SSA_stept &step) { return step.ignore; });`?

File: src/goto-symex/symex_target_equation.h [hunk 3]
```diff
@@ -391,10 +387,8 @@ class symex_target_equationt:public symex_targett
...
   {
-    for(SSA_stepst::const_iterator it=SSA_steps.begin();
-        it!=SSA_steps.end();
-        it++)
-      if(it->source.thread_nr!=0)
+    for(const auto &step: SSA_steps)
+      if(step.source.thread_nr!=0)
         return true;
 
     return false;
```
Comment (tautschnig) [inline src/goto-symex/symex_target_equation.h]: How about `std::any_of(SSA_steps.begin(), SSA_steps.end(), [](const SSA_stept &step) { return step.source.thread_nr != 0; });`?

**Review by tautschnig — APPROVED** [review]

**Comment by kroening** [comment]
All done

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7b064a70cd9f516bbf7549a03661ad29fed416ea).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97926682

**Pull request merged**